sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java

```diff
@@ -469,22 +475,39 @@ public IterableStream<ChatCompletions> getChatCompletionsStream(
      * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
      * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
      * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return the {@link ImageOperationResponse} for polling of status details for long running operations.
+     * @return the {@link ImageResponse} for the image generation result.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public ImageResponse generateImage(ImageGenerationOptions imageGenerationOptions) {
+    public ImageResponse getImages(ImageGenerationOptions imageGenerationOptions) {
         RequestOptions requestOptions = new RequestOptions();
         BinaryData imageGenerationOptionsBinaryData = BinaryData.fromObject(imageGenerationOptions);
-        return openAIServiceClient != null
-                ? openAIServiceClient
-                        .generateImageWithResponse(imageGenerationOptionsBinaryData, requestOptions)
-                        .getValue()
-                        .toObject(ImageResponse.class)
-                : beginBeginAzureBatchImageGeneration(imageGenerationOptionsBinaryData, requestOptions)
-                        .waitForCompletion()
-                        .getValue()
-                        .toObject(ImageOperationResponse.class)
-                        .getResult();
+
+        if (openAIServiceClient != null) {
+            return openAIServiceClient
+                .generateImageWithResponse(imageGenerationOptionsBinaryData, requestOptions)
+                .getValue()
+                .toObject(ImageResponse.class);
+        } else {
+            // TODO: Currently, we use async client block() to avoid a unknown LRO status "notRunning" which Azure Core will
+            //       fix the issue in August release and we will reuse the method
+            //       "SyncPoller<BinaryData, BinaryData> beginBeginAzureBatchImageGeneration()" after.
+            try {
+                return this.serviceClient.beginBeginAzureBatchImageGenerationAsync(imageGenerationOptionsBinaryData,
+                        requestOptions)
+                    .last()
+                    .flatMap(it -> it.getFinalResult())
+                    .map(it -> it.toObject(ImageOperationResponse.class).getResult()).block();
+            } catch (Exception e) {
+                Throwable unwrapped = Exceptions.unwrap(e);
+                if (unwrapped instanceof RuntimeException) {
+                    throw LOGGER.logExceptionAsError((RuntimeException) unwrapped);
+                } else if (unwrapped instanceof IOException) {
+                    throw LOGGER.logExceptionAsError(new UncheckedIOException((IOException) unwrapped));
+                } else {
+                    throw LOGGER.logExceptionAsError(new RuntimeException(unwrapped));
+                }
```

Comment:
Could you explain what `Exceptions.unwrap` if for and the usage of `LOGGER` here? I am not entirely sure I understand why this is done like this. I though you could simply do:

```java
try {
  // blocking call
} catch (Exception e) {
  throw new RuntimeException(e);
}
```